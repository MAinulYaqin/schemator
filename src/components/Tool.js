import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import Tooltip from "./Tooltip";

const StyledTooltip = styled(Tooltip)``;

const Container = styled.div`
  width: 32px;
  height: 32px;
  box-sizing: border-box;
  border-radius: 4px;
  background: #555;
  margin: 4px;
  display: flex;
  cursor: pointer;
  user-select: none;
  position: relative;

  &:hover {
    background: "#eee";
  }

  &:hover ${StyledTooltip} {
    opacity: 1;
  }
`;

const Tool = ({ icon, tooltip, isDisabled, onClick }) => {
  const StyledIcon = styled(icon)`
    color: ${({ isDisabled }) => (isDisabled ? "#666" : "#aaa")};
    font-size: 24px;
    margin: auto;
  `;

  return (
    <Container onClick={() => !isDisabled && onClick()} isDisabled={isDisabled}>
      {tooltip && <StyledTooltip text={tooltip} />}
      <StyledIcon name={icon} isDisabled={isDisabled} />
    </Container>
  );
};

Tool.propTypes = {
  icon: PropTypes.func.isRequired,
  tooltip: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  isDisabled: PropTypes.bool
};

export default Tool;
