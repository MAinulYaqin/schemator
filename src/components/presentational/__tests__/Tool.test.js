import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import IconSave from "react-icons/lib/md/save";

import Tool from "../Tool";

describe("Tool", () => {
  const setup = propOverrides => {
    const props = {
      tooltip: "Save Schema",
      icon: IconSave,
      onClick: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<Tool {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should renders as disabled", () => {
    const { wrapper } = setup({ isDisabled: true });

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should calls 'onClick' while clicked", () => {
    const { wrapper, props } = setup();

    wrapper.simulate("click");

    expect(props.onClick).toBeCalled();
  });

  it("should not calls 'onClick' while clicked in disable", () => {
    const { wrapper, props } = setup({ isDisabled: true });

    wrapper.simulate("click");

    expect(props.onClick).not.toBeCalled();
  });
});
