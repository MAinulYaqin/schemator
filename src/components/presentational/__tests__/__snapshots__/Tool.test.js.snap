// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tool should renders default 1`] = `
<styled.div
  onClick={[Function]}
>
  <Styled(Tooltip)
    text="Save Schema"
  />
  <Styled(MdSave)
    color="#999"
    size={24}
  />
</styled.div>
`;

exports[`Tool should renders in active 1`] = `
<styled.div
  onClick={[Function]}
>
  <Styled(Tooltip)
    text="Save Schema"
  />
  <Styled(MdSave)
    color="white"
    size={24}
  />
</styled.div>
`;

exports[`Tool should renders without tooltip 1`] = `
<styled.div
  onClick={[Function]}
>
  <Styled(MdSave)
    color="#999"
    size={24}
  />
</styled.div>
`;
